gossip/client/namedtree.go

```diff
@@ -0,0 +1,113 @@
+package client
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"strings"
+
+	"github.com/quorumcontrol/chaintree/chaintree"
+	"github.com/quorumcontrol/chaintree/nodestore"
+	"github.com/quorumcontrol/messages/v2/build/go/transactions"
+	"github.com/quorumcontrol/tupelo-go-sdk/consensus"
+)
+
+type Generator struct {
+	Namespace string
+	Client    *Client
+}
+
+type NamedChainTree struct {
+	Name      string
+	ChainTree *consensus.SignedChainTree
+	Client    *Client
+
+	genesisKey *ecdsa.PrivateKey
+	nodeStore  nodestore.DagStore
+	owners     []string
+}
+
+type NamedChainTreeOptions struct {
+	Name              string
+	ObjectStorageType string
+	Client            *Client
+	NodeStore         nodestore.DagStore
+	Owners            []string
+}
+
+// GenesisKey creates a new named key creation key based on the supplied name.
+// It lower-cases the name first to ensure that chaintree names are case
+// insensitive.
+func (g *Generator) GenesisKey(name string) (*ecdsa.PrivateKey, error) {
+	// TODO: If we ever want case sensitivity, consider adding a bool flag
+	// to the Generator or adding a new fun.
+	lowerCased := strings.ToLower(name)
+	return consensus.PassPhraseKey([]byte(lowerCased), []byte(g.Namespace))
+}
+
+func (g *Generator) New(ctx context.Context, opts *NamedChainTreeOptions) (*NamedChainTree, error) {
```

Comment:
I don't think `New` implies a side effect. Maybe a separate method for "generate" or call this `Create` ?